yletunnus/tests/test_backend.py

```diff
@@ -91,18 +95,46 @@ def authorize_body(self, request, url, headers):
 
     def auth_handlers(self, start_url):
         target_url = super().auth_handlers(start_url)
+        if self.strategy.setting('YLETUNNUS_ENVIRONMENT') == 'production':
+            assert start_url.startswith('https://auth.api.yle.fi/authorize')
+        elif self.strategy.setting('YLETUNNUS_ENVIRONMENT') == 'test':
+            assert start_url.startswith('https://auth.api-test.yle.fi/authorize')
         self.complete_url = target_url
         HTTPretty.register_uri(HTTPretty.GET, start_url, body=self.authorize_body)
         return target_url
 
     def test_login(self):
-        self.strategy.set_settings({
-            'SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL': True
-        })
-        self.do_login()
+        for environment in VALID_ENVIRONMENTS:
+            with self.subTest():
+                self.strategy.set_settings({
+                    'YLETUNNUS_ENVIRONMENT': environment,
+                    'SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL': True
+                })
+                self.do_login()
+
+    def test_login_invalid_environment_raises_error(self):
+        for environment in INVALID_ENVIRONMENTS:
+            with self.subTest():
+                self.strategy.set_settings({
+                    'YLETUNNUS_ENVIRONMENT': environment,
+                    'SOCIAL_AUTH_USERNAME_IS_FULL_EMAIL': True
+                })
+                try:
+                    self.do_login()
```

Comment:
Could this be clearer as something like this?

```python
with pytest.raises(ImproperlyConfigured):
    self.do_login()
```

Reply:
Ah. Sorry this was a unittest class. Maybe should use [TestCase.assertRaises](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises)?

Reply:
But aren't these tests still run with pytest? And pytest supports unittest classes, so can use `with pytest.raises`?

Reply:
Of course one _can_, but I think it's neater to not mix and match unittest and pytest stuff. No need to import pytest then either.

Reply:
> These new fix commits could be squashed to the original commit?

I will, in the end.

Reply:
564e6e929356dd12285e052a1131813f8b8e41de fixes it with pytest, but I can do it with TestCase also.

Reply:
Fixed in 4bd071ad16607afa64bf7b14a09324b33a49b167